src/python/gem5/resources/resource.py

```diff
@@ -554,6 +555,70 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    def __init__(
+        self,
+        workloads: List[Dict[str, Any]],
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            local_path=local_path,
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+        self.resource_version = resource_version
+        self.workloads = workloads
+        self.workload_obj_list = []
+        self.index = 0
+        self._local_path = local_path
+        self._description = description
+        self._source = source
+        self._obtain_workloads_list()
+
+    def _obtain_workloads_list(self) -> None:
+        for workload in self.workloads:
+            self.workload_obj_list.append(
+                WorkloadResource(
```

Comment:
Is there a reason we don't use "obtain_resource" instead? This isn't getting all the information about a resource we may need.

Reply:
Yes, we should be using `obtain_resource`. For someone who wants to create their own `SuiteResource` how do we handle the call to `obtain_resource`? I can think of 2 possible solutions:

- We ask users to create their own `resources.json` and pass it in as a client in ontain_resources, We can also pass the `clients` parameter to the `SuiteResource` constructor so that users can define which clients they want.

- We can make getters and setters for the `workload_obj_list` to allow users to directly edit the workload list. This needs some sort of way to not call `obtain_resources` when the user is using their resources. 

I am leaning towards the first one.